src/main/java/chess/domain/Position.java

```diff
@@ -0,0 +1,227 @@
+package chess.domain;
+
+import chess.domain.board.ChessBoard;
+import chess.domain.move.Direction;
+import chess.domain.move.KnightType;
+import chess.domain.move.MoveType;
+import chess.domain.piece.Pawn;
+
+import java.util.Objects;
+
+public class Position {
+    private static final String ERROR_MESSAGE_EXIST_PIECE_ON_PATH = "경로에 다른 말이 존재합니다.";
+    private static final int Y_POSITION_VALUE = 1;
+    private static final int X_POSITION_INDEX = 0;
+
+    private XPosition xPosition;
+    private YPosition yPosition;
+
+    private Position(XPosition xPosition, YPosition yPosition) {
+        this.xPosition = xPosition;
+        this.yPosition = yPosition;
+    }
+
+    public static Position of(String coordinate) {
+        String[] xyPositionValue = coordinate.split("");
+        return new Position(XPosition.of(xyPositionValue[X_POSITION_INDEX])
+                , YPosition.of(xyPositionValue[Y_POSITION_VALUE]));
+    }
+
+    public static Position of(XPosition xPosition, YPosition yPosition) {
+        return new Position(xPosition, yPosition);
+    }
+
+    public boolean isNewLine() {
+        return xPosition == XPosition.A;
+    }
+
+    public boolean isSameRank(Position target) {
+        return this.yPosition == target.yPosition;
+    }
+
+    public boolean isSameFile(Position target) {
+        return isSameFile(target.xPosition);
+    }
+
+    public boolean isSameFile(XPosition xPosition) {
+        return this.xPosition == xPosition;
+    }
+
+    public boolean isUpDirection(Position target) {
+        return calculateYPositionDistance(target) < 0;
+    }
+
+    public boolean isDownDirection(Position target) {
+        return calculateYPositionDistance(target) > 0;
+    }
+
+    public boolean isLeftDirection(Position target) {
+        return calculateXPositionDistance(target) > 0;
+    }
+
+    public boolean isRightDirection(Position target) {
+        return calculateXPositionDistance(target) < 0;
+    }
+
+    public int calculateYPositionDistance(Position target) {
+        return yPosition.getNumber() - target.yPosition.getNumber();
+    }
+
+    public int calculateXPositionDistance(Position target) {
+        return xPosition.getNumber() - target.xPosition.getNumber();
+    }
+
+    private void isExistPieceOnPath(ChessBoard chessBoard, int xDegree, int yDegree) {
+        if (chessBoard.isExistPiece(Position.of(XPosition.of(xDegree), YPosition.of(yDegree)))) {
```

Comment:
```suggestion
        if (chessBoard.isExistPiece(Position.of(xDegree, yDegree))) {
```

Position에서 해당 생성자를 제공해주면 가독성이 상승할 것 같습니다.